internal/service/rds/engine_version_data_source_test.go

```diff
@@ -107,6 +107,25 @@ func TestAccRDSEngineVersionDataSource_defaultOnly(t *testing.T) {
 	})
 }
 
+func TestAccRDSEngineVersionDataSource_filters(t *testing.T) {
+	dataSourceName := "data.aws_rds_engine_version.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t); testAccEngineVersionPreCheck(t) },
+		ErrorCheck:               acctest.ErrorCheck(t, rds.EndpointsID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             nil,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccEngineVersionDataSourceConfig_filters(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(dataSourceName, "version"),
+				),
```

Comment:
The test check function here does not appear to validate the features being added - it's only checking for the DB version instead of validating that only serverless engine modes are being returned. Using the HCL provided as an example, a good test would validate the engine type matches `aurora` and returns a values of `serverless` for the engine mode.